opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -145,6 +145,62 @@ def from_map(cls, maxlen, mapping):
 Link = namedtuple("Link", ("context", "attributes"))
 
 
+class SpanProcessor:
+    def on_start(self, span: "Span") -> None:
```

Comment:
Shall we mention this is a SDK `Span`, not a API `Span`?

Reply:
I'll mention this explicitly in a the `SpanProcessor` docs.

I had tried to move it there but `Span` needs `_NO_OP_SPAN_PROCESSOR` that is an instance of `Span`, so I gave up and preferred to use strings for the annotations.

Reply:
Sorry if this is covered somewhere, but why would the SpanProcessor take the SDK Span as an object? Shouldn't SpanProcessors be more agnostic of the span implementation?

Reply:
@toumorokoshi We specify SpanProcessors only in the SDK level, so there can't ever be something different than an SDK span. Also, a plain span would be pretty useless to a span processor since there aren't any properties on it that can be read.

Reply:
@toumorokoshi as already pointed by @Oberon00 the SpanProcessors are only defined in the SDK level.  We discussed in the SIG meeting about passing a readable span (something similar to SpanData) to the SpanProcessors, but we then agreed to pass directly the SDK span for the time being.

Reply:
And that in the SIG was only about exporters, where this might change in the long term. I imagine SpanProcessors will continue to always just receive a sdk.Span object directly.

Reply:
> I imagine SpanProcessors will continue to always just receive a sdk.Span object directly.

Correct. It has been mentioned the (future) possibility to set properties on the SDK `Span` from processor code ;)